### PR TITLE
abdullah nodejs week2 hw

### PR DESCRIPTION
**Pull request by heartwhite**

### CODE REVIEW

File: week2/homework/src/remove.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const desc = 'REMOVE: removes list item with 1-based index';
+const toDos = fs.readFileSync('toDoList.txt', 'utf8');
+function remove(index) {
+  if (Number.isInteger(Number(index)) && index > 0) {
+    const toDosArr = toDos.split('\n');
+    const spliced = toDosArr.splice(index - 1, 1);
+    const toDosJoined = toDosArr.join('\n');
+    fs.writeFile('toDoList.txt', toDosJoined, err => {
+      if (err) {
+        console.log(err);
```
Comment (gajduk) [inline week2/homework/src/remove.js]: Error handling 👍

File: week2/homework/src/list.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const fs = require('fs');
+const desc = 'LIST: shows the toDoList';
+const toDos = fs.readFileSync('toDoList.txt', 'utf8');
```
Comment (gajduk) [inline week2/homework/src/list.js]: What would happen if there is no such file?

File: week2/homework/src/update.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    const toDosJoined = toDosArr.join('\n');
+    fs.writeFile('toDoList.txt', toDosJoined, err => {
+      if (err) {
+        console.log(err);
+      } else {
+        console.log(`${index}. item ( ${oldToDo} ) updated to "${newToDo}"`);
+      }
+    });
+  } else {
+    console.log('WARN: first parameter of update command must be an integer and greater than 0!');
```
Comment (gajduk) [inline week2/homework/src/update.js]: consistent error handling 👍

File: week2/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,56 @@
...
+const help = require('./help.js');
+const command = process.argv[2];
+const invalidTextMessage = 'WARN: nothing to add ! type sth for your toDoList!';
 
-// TODO: Write the homework code in this file
+switch (command) {
+  case 'add':
+    if (process.argv[3]) {
+      if (process.argv[4]) {
+        console.log('WARN: please type the toDo item without spaces. you can use dash ( - )');
```
[the remark this answers is not in the document]
Comment (heartwhite) [reply week2/homework/src/index.js]: ok. no limitation and no dashes. its done.!

File: week2/homework/src/list.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (heartwhite) [reply week2/homework/src/list.js]: inside. you re right. I changed it

Comment (heartwhite) [reply week2/homework/src/list.js]: I decided to create an empty txt when the js file called (node .)

File: week2/homework/src/update.js (hunk 3 above)
Comment (heartwhite) [reply week2/homework/src/update.js]: thanks.

File: week2/homework/src/remove.js (hunk 1 above)
Comment (heartwhite) [reply week2/homework/src/remove.js]: 👍

File: week2/homework/src/index.js [hunk 5]
```diff
@@ -1,3 +1,56 @@
 'use strict';
+const fs = require('fs');
+const add = require('./add.js');
+const reset = require('./reset.js');
+const list = require('./list.js');
+const remove = require('./remove.js');
+const update = require('./update.js');
+const help = require('./help.js');
```
[the remark this answers is not in the document]
Comment (heartwhite) [reply week2/homework/src/index.js]: 👍

File: week2/homework/src/help.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const fs = require('fs');
+const desc = 'HELP : shows command list with small descriptions';
+function help(array) {
+  array.forEach(element => {
```
[the remark this answers is not in the document]
Comment (heartwhite) [reply week2/homework/src/help.js]: 👍

File: week2/homework/src/add.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+const fs = require('fs');
+const desc = 'adds the text you type after command word to toDoList';
```
[the remark this answers is not in the document]
Comment (heartwhite) [reply week2/homework/src/add.js]: 👍

**Comment by heartwhite** [comment]
" no such file or directory" bug fixed.

**Pull request closed without merging**